### DISCUSSION TITLE
V3 dependencies changes

### DISCUSSION THREAD

**Discussion by madalinignisca** [area17/twill#1180] [Ideas]
Hi.

For v3, can we make some of the dependencies optional and help the developer to install them if he really needs and will use them?

To give a list:
- doctrine/dbal
- laravel/socialite
- laravel/ui (do not even understand why would this be needed)
- league/flysystem-aws-s3-v3 (really not needed for non-aws people...)
- matthewbdaly/laravel-azure-storage (hated by non-azure people)
- pragmarx/google2fa-qrcode (require if 2fa will be used!)
- spatie/laravel-analytics (it even made some other helpers cry because it is installed)

A lot of dependencies should be installed only and only if the user will be using them.

It makes sense even for production deployments where we squeeze container images to be small as possible, and care about updates download bandwidth...

**Comment by ifox** [comment]
Hi @madalinignisca, you're totally right and I confirm this is something we had in mind for Twill 3.
